guides/source/active_record_callbacks.md

```diff
@@ -524,68 +884,33 @@ Callbacks can mix both `:if` and `:unless` in the same declaration:
-
-```ruby
-class FileDestroyerCallback
-  def after_destroy(file)
-    if File.exist?(file.filepath)
-      File.delete(file.filepath)
-    end
-  end
-end
-```
-
-When declared inside a class, as above, the callback methods will receive the model object as a parameter. This will work on any model that uses the class like so:
-
-```ruby
-class PictureFile < ApplicationRecord
-  after_destroy FileDestroyerCallback.new
-end
-```
-
-Note that we needed to instantiate a new `FileDestroyerCallback` object, since we declared our callback as an instance method. This is particularly useful if the callbacks make use of the state of the instantiated object. Often, however, it will make more sense to declare the callbacks as class methods:
-
-```ruby
-class FileDestroyerCallback
-  def self.after_destroy(file)
-    if File.exist?(file.filepath)
-      File.delete(file.filepath)
-    end
-  end
-end
-```
-
-When the callback method is declared this way, it won't be necessary to instantiate a new `FileDestroyerCallback` object in our model.
-
-```ruby
-class PictureFile < ApplicationRecord
-  after_destroy FileDestroyerCallback
-end
-```
-
-You can declare as many callbacks as you want inside your callback classes.
 
 Transaction Callbacks
 ---------------------
 
 ### `after_commit` and `after_rollback`
 
-There are two additional callbacks that are triggered by the completion of a database transaction: [`after_commit`][] and [`after_rollback`][]. These callbacks are very similar to the `after_save` callback except that they don't execute until after database changes have either been committed or rolled back. They are most useful when your Active Record models need to interact with external systems which are not part of the database transaction.
+Two additional callbacks are triggered by the completion of a database
+transaction: [`after_commit`][] and [`after_rollback`][]. These callbacks are
+very similar to the `after_save` callback except that they don't execute until
+after database changes have either been committed or rolled back. They are most
+useful when your Active Record models need to interact with external systems
+that are not part of the database transaction.
 
-Consider, for example, the previous example where the `PictureFile` model needs to delete a file after the corresponding record is destroyed. If anything raises an exception after the `after_destroy` callback is called and the transaction rolls back, the file will have been deleted and the model will be left in an inconsistent state. For example, suppose that `picture_file_2` in the code below is not valid and the `save!` method raises an error.
+Consider if the `PictureFile` model, from the previous example, needs to delete
```

Comment:
Since these have been moved around, there's no "previous" example for the PictureFile, so we'll have to introduce it here.

Reply:
Good catch! I introduced it and changed the way the Callback objects example works a little bit. Can you please double check that everything looks good?